specs/WebMessageObjects.md

```diff
@@ -119,6 +138,64 @@ void WebView_WebMessageReceivedHandler(object sender, CoreWebView2WebMessageRece
 }
 ```
 
+## Post a FileSystemHandle to the web content
+Use this API to create a FileSystemHandle from the app and send it to the web content in WebView2.
+
+### Web content code
+```javascript
+chrome.webview.addEventListener("message", function (e) {
+    if (e.data.MyMessageType === "FileHandle") {
+        var fileHandle = e.additionalObjects[0];
+        if (fileHandle.kind === "file") {
+            // run file code
+        } else if (fileHandle.kind === "directory") {
+            // run directory code
+        }
+    }
+});
+```
+
+### App code
+```cpp
+wil::com_ptr<ICoreWebView2_21> webview21 =
+    m_webView2.try_query<ICoreWebView2_21>();
+wil::com_ptr<ICoreWebView2Environment> environment =
+    appWindow->GetWebViewEnvironment();
+wil::com_ptr<ICoreWebView2Environment14> environment14 = 
+    environment.try_query<ICoreWebView2Environment14>();
+wil::com_ptr<ICoreWebView2FileSystemHandle> fileHandle;
+CHECK_FAILURE(environment14->CreateWebFileSystemHandle(

```

Comment:
Should be more explicit what the behavior of this is. It sounds from later comments like it's an open-if-exists operation (not fail-if-exists, not overwrite).

And what exclusive sharing?